pkg/sharding/apiserver/handler.go

```diff
@@ -0,0 +1,451 @@
+		)
+		return
+	}
+	tableConverter := storage.TableConvertor{TableGenerator: tableGenerator}
+
+	nopAuthorizer := authorizer.AuthorizerFunc(func(ctx context.Context, a authorizer.Attributes) (authorizer.Decision, string, error) {
+		return authorizer.DecisionAllow, "", nil
+	})
+
+	requestScope := &handlers.RequestScope{
+		Namer: nopScopeNamer{
+			selfLinker: meta.NewAccessor(),
+		},
+		Serializer:                  negotiatedSerializer,
+		ParameterCodec:              parameterCodec,
+		StandardSerializers:         negotiatedSerializer.SupportedMediaTypes(),
+		Creater:                     apiserver.UnstructuredCreator{},
+		Convertor:                   nopConverter{},
+		Defaulter:                   nopDefaulter{},
+		Typer:                       crdserverscheme.NewUnstructuredObjectTyper(),
+		UnsafeConvertor:             nopConverter{},
+		Authorizer:                  nopAuthorizer,
+		EquivalentResourceMapper:    equivalentResourceRegistry,
+		TableConvertor:              tableConverter,
+		Resource:                    resource,
+		Subresource:                 requestInfo.Subresource,
+		MetaGroupVersion:            metav1.SchemeGroupVersion,
+		HubGroupVersion:             groupVersion,
+		MaxRequestBodyBytes:         h.maxRequestBodyBytes,
+		AcceptsGroupVersionDelegate: everythingAcceptor{},
+	}
+
+	nopAdmission := admission.NewHandler()
+
+	userInfo, ok := request.UserFrom(req.Context())
+	if !ok {
+		responsewriters.ErrorNegotiated(
+			apierrors.NewInternalError(fmt.Errorf("missing userInfo")),
+			codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
+		)
+		return
+	}
+
+	// some handlers like patch end up reading the body before we get a chance to handle things, so
+	// we need to deep-copy the request as best we can so that we can forward the requests along
+	// TODO: is there a better way?
+	cloned := net.CloneRequest(req)
+	// copied from DumpRequest
+	if req.Body == nil || req.Body == http.NoBody {
+		// No copying needed. Preserve the magic sentinel meaning of NoBody.
+		req.Body = http.NoBody
+		cloned.Body = http.NoBody
+	} else {
+		var buf bytes.Buffer
+		if _, err := buf.ReadFrom(req.Body); err != nil {
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("failed reading request body")),
+				codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
+			)
+			return
+		}
+		if err := req.Body.Close(); err != nil {
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("failed closing request body")),
+				codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
+			)
+			return
+		}
+		req.Body = io.NopCloser(&buf)
+		cloned.Body = io.NopCloser(bytes.NewReader(buf.Bytes()))
+	}
+
+	klog.Info("v=== INCOMING ===v")
+	v, e := httputil.DumpRequest(cloned, true)
+	klog.Info(string(v))
```

Comment:
might be too verbose

Reply:
Yeah, I have a commit that pushes these to V(10) I will un-revert before merge